ha/core/support_bundle/ha_bundle.py

```diff
@@ -0,0 +1,104 @@
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import os
+
+from cortx.utils.schema.payload import Tar
+from cortx.utils.process import SimpleProcess
+from cortx.utils.log import Log
+
+from ha import const
+from ha.core.error import SupportBundleError
+
+class HABundle:
+
+    def __init__(self):
+        """
+        This Class generates the Support Bundle for Component HA.
+        This method will generate bundle for CSM and include the logs in it.
+        :param command: Csm_cli Command Object :type: command
+        :return:
+        """
+        # TODO Add setup detection logic
+        self._setup_type = "ldr"
+        self._id = None
+        self._path = None
+
+    def process_request(self, action, args, output):
+        """
+        Process support Bundle request
+
+        Args:
+            action (string): Action to be taken for support bundle.
+            args (dict): Parameter for support bundle
+        """
+        self._output = output
+        self._id = args.id
+        self._path = args.path
+        getattr(self, args.bundle_action)()
+
+    def create(self):
+        """
+        Create support bundle on given path.
+        """
+        if self._setup_type == "ldr":
+            self._add_more_log()
+        export_path = self._export_logs(self._check_files())
+        self._output.output(f"Successfully created support bundle at {export_path}")
+
+    def _add_more_log(self):
+        """
+        Collect Extra Log
+        """
+        cmd_list = [const.PCS_STATUS, const.PCS_FAILCOUNT_STATUS]
+        with open(const.EXTRA_CMD_LOG, "w") as file:
+            for cmd in cmd_list:
+                _err = ""
+                _proc = SimpleProcess(cmd)
+                _output, _err, _rc = _proc.run(universal_newlines=True)
+                file.write("\n\n\n********************************\n")
+                file.write(f"Command: {cmd}\nReturn Code: {_rc}\nOutput:\n")
+                file.write(f"{_output}")
+                file.write(f"Error if any: {_err}\n\n\n")
+
+    def _check_files(self):
+        """
+        Check all support bundle file for HA
+        and provide list of available file
+        """
+        ha_logs = list(const.SUPPORT_BUNDLE_LOGS)
+        for _path in ha_logs:
+            if not os.path.exists(_path):
+                ha_logs.remove(_path)
+                Log.warn(f"For support bundle {_path} is not available")
```

Comment:
We should create support_bundle.err file, add these logs to the file and make it part of support bundle. Same file may be used for dumping error in _add_more_logs functions too.

Reply:
Done